interface/wx/webrequest.h

```diff
@@ -0,0 +1,639 @@
+
+        The next request will be a HTTP @c POST instead of the default HTTP
+        @c GET and the given @a dataStream will be posted as the body of
+        this request.
+
+        @param dataStream
+            The data in this stream will be posted as the request body
+        @param contentType
+            The value of HTTP "Content-Type" header, e.g.
+            "application/octet-stream".
+        @param dataSize
+            Amount of data which is sent to the server. If set to
+            @c wxInvalidOffset all stream data is sent.
+
+    */
+    void SetData(wxSharedPtr<wxInputStream> dataStream,
+        const wxString& contentType, wxFileOffset dataSize = wxInvalidOffset);
+
+    /**
+        Instructs the request to ignore server error status codes.
+
+        Per default, server side errors (status code 400-599) will enter
+        the State_Failed state just like network errors, but
+        if the response is still required in such cases (e.g. to get more
+        details from the response body), set this option to ignore all errors.
+        If ignored, wxWebResponse::GetStatus() has to be checked
+        from the State_Completed event handler.
+    */
+    void SetIgnoreServerErrorStatus(bool ignore);
+
+    /**
+        Sets how response data will be stored.
+
+        The default storage method @c Storage_Memory collects all response data
+        in memory until the request is completed. This is fine for most usage
+        scenarios like API calls, loading images, etc. For larger downloads or
+        if the response data will be used permanently @c Storage_File instructs
+        the request to write the response to a temporary file. This temporary
+        file may then be read or moved after the request is complete. The file
+        will be downloaded to the system temp directory as returned by
+        wxStandardPaths::GetTempDir(). To specify a different directory use
+        wxWebSession::SetTempDir().
+
+        Sometimes response data needs to be processed while its downloaded from
+        the server. For example if the response is in a format that can be
+        parsed piece by piece like XML, JSON or an archive format like ZIP.
+        In these cases storing the data in memory or a file before being able
+        to process it might not be ideal and @c Storage_None should be set.
+        With this storage method the data is only available during the
+        @c wxEVT_WEBREQUEST_DATA event calls as soon as it's received from the
+        server.
+    */
+    void SetStorage(Storage storage);
+    ///@}
+
+    /** @name Progress methods
+        Methods that describe the requests progress
+    */
+    ///@{
+    /**
+        Returns the current state of the request.
+    */
+    State GetState() const;
+
+    /// Returns the number of bytes sent to the server.
+    wxFileOffset GetBytesSent() const;
+
+    /// Returns the number of bytes expected to be send to the server.
```

Comment:
```suggestion
    /// Returns the number of bytes expected to be sent to the server.
```
Could you please clarify in this comment if this is the total number of the remaining number of bytes to be sent?